lib/msal-common/src/authority/Authority.ts

```diff
@@ -272,40 +273,32 @@ export class Authority {
             return AuthorityMetadataSource.CACHE;
         }
 
+        let harcodedMetadata = this.getEndpointMetadataFromHardcodedValues();
         metadata = await this.getEndpointMetadataFromNetwork();
         if (metadata) {
             // If the user prefers to use an azure region replace the global endpoints with regional information.
             if (this.authorityOptions.azureRegionConfiguration?.azureRegion) {
-                const autodetectedRegionName = await this.regionDiscovery.detectRegion(this.authorityOptions.azureRegionConfiguration.environmentRegion, this.regionDiscoveryMetadata, this.proxyUrl);
-
-                const azureRegion = this.authorityOptions.azureRegionConfiguration.azureRegion === Constants.AZURE_REGION_AUTO_DISCOVER_FLAG
-                    ? autodetectedRegionName
-                    : this.authorityOptions.azureRegionConfiguration.azureRegion;
-
-                if (this.authorityOptions.azureRegionConfiguration.azureRegion === Constants.AZURE_REGION_AUTO_DISCOVER_FLAG) {
-                    this.regionDiscoveryMetadata.region_outcome = autodetectedRegionName ?
-                        RegionDiscoveryOutcomes.AUTO_DETECTION_REQUESTED_SUCCESSFUL :
-                        RegionDiscoveryOutcomes.AUTO_DETECTION_REQUESTED_FAILED;
-                } else {
-                    if (autodetectedRegionName) {
-                        this.regionDiscoveryMetadata.region_outcome = (this.authorityOptions.azureRegionConfiguration.azureRegion === autodetectedRegionName) ?
-                            RegionDiscoveryOutcomes.CONFIGURED_MATCHES_DETECTED :
-                            RegionDiscoveryOutcomes.CONFIGURED_NOT_DETECTED;
-                    } else {
-                        this.regionDiscoveryMetadata.region_outcome = RegionDiscoveryOutcomes.CONFIGURED_NO_AUTO_DETECTION;
-                    }
-                }
-
-                if (azureRegion) {
-                    this.regionDiscoveryMetadata.region_used = azureRegion;
-                    metadata = Authority.replaceWithRegionalInformation(metadata, azureRegion);
-                }
+                metadata = await this.updateMetadataWithRegionalInformation(metadata);
             }
 
             metadataEntity.updateEndpointMetadata(metadata, true);
             return AuthorityMetadataSource.NETWORK;
+        }    
+
+        if (harcodedMetadata && !this.authorityOptions.skipAuthorityMetadataCache) {
```

Comment:
I don't see new unit tests that test this code path.

Reply:
This is where `instance_aware` should probably be checked as well. I would think that if instance_aware is enabled, we should not use the hardcoded metadata.